sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/utils/JsonConverterUtil.java

```diff
@@ -3,61 +3,95 @@
 
 package com.azure.security.keyvault.jca.implementation.utils;
 
-import com.fasterxml.jackson.core.JsonProcessingException;
-import com.fasterxml.jackson.databind.DeserializationFeature;
-import com.fasterxml.jackson.databind.ObjectMapper;
+import com.azure.json.JsonProviders;
+import com.azure.json.JsonReader;
+import com.azure.json.JsonSerializable;
+import com.azure.json.JsonWriter;
 
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.logging.Logger;
 
 import static java.util.logging.Level.WARNING;
 
 /**
- * The Jackson JsonConverter.
+ * The JSON Converter.
  */
 public final class JsonConverterUtil {
-    private static final ObjectMapper FROM_JSON_MAPPER = new ObjectMapper()
-        .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
-    private static final ObjectMapper TO_JSON_MAPPER = new ObjectMapper();
-
     /**
      * Stores the logger.
      */
     private static final Logger LOGGER = Logger.getLogger(JsonConverterUtil.class.getName());
 
     /**
-     * From JSON.
+     * Stores the JSON cache.
+     */
+    private static final Map<Class<?>, ReflectiveInvoker> FROM_JSON_CACHE;
+
+    static {
+        FROM_JSON_CACHE = new ConcurrentHashMap<>();
```

Comment:
We don't really need this in JCA as the set of types being deserialized is known as they're all defined by the package itself (unlike in Core which doesn't know what it's deserializing other than it's a `JsonSerializable` type). Let's remove this and change the deserialization method to `public static Object fromJson(String json, Function<JsonReader, Object> deserializationFunction)`. Also, I think we can change `Object` to `T`.